### ISSUE TITLE
resource/aws_backup_selection: IAM Role cannot be assumed by AWS Backup

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#9297]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.3
+ provider.aws v2.18.0
```

### Affected Resource(s)

* aws_backup_selection

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    aws = "2.18.0"
  }
}

provider "aws" {
  region = "us-east-2"
}

data "aws_caller_identity" "current" {}

data "aws_region" "current" {}

resource "aws_iam_role" "test" {
  name               = "9269-testing"
  assume_role_policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": ["sts:AssumeRole"],
      "Effect": "allow",
      "Principal": {
        "Service": ["backup.amazonaws.com"]
      }
    }
  ]
}
POLICY
}

resource "aws_iam_role_policy_attachment" "test" {
  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSBackupServiceRolePolicyForBackup"
  role = "${aws_iam_role.test.name}"
}

resource "aws_backup_vault" "test" {
  name = "9269-testing"
}

resource "aws_backup_plan" "test" {
  name = "9269-testing"

  rule {
    rule_name = "test"
    target_vault_name = "${aws_backup_vault.test.name}"
    schedule = "cron(0 12 * * ? *)"
  }
}

resource "aws_backup_selection" "test" {
  iam_role_arn = "${aws_iam_role.test.arn}"
  name = "9269-testing"
  plan_id = "${aws_backup_plan.test.id}"

  resources = [
    "arn:aws:ec2:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:volume/*",
  ]
}
```

### Expected Behavior

Successful AWS Backup Selection creation.

### Actual Behavior

```
Error: error creating Backup Selection: InvalidParameterValueException: IAM Role arn:aws:iam::--OMITTED--:role/9269-testing cannot be assumed by AWS Backup
```

### Steps to Reproduce

1. `terraform init`
1. `terraform apply`

### Important Factoids

Retrying `terraform apply` works. Appears to be IAM eventual consistency issue.

**Comment by bflad** [comment]
Bug fix submitted: #9298

**Comment by nywilken** [comment]
The fix to the `aws_backup_selection` resource has been merged and will be released with version 2.19.0 of the Terraform AWS provider.

**Comment by bflad** [comment]
This has been released in [version 2.19.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2190-july-11-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!

**Comment by tbugfinder** [comment]
I'm wondering if anybody else could execute above code using tf 0.11.14 or tf 0.12.10 ?

In my case it complains about the `*`.
